### PR TITLE
refactor(graphql): GraphQL Public API Refactor + Documentation

### PR DESCRIPTION
**Pull request by jjoyce0510**
This PR refactors the GraphQL API mainly addressing conventional consistency concerns. It also adds a bunch of new GraphQL API documentation! Rendered: 
- [Overview](https://github.com/linkedin/datahub/blob/bbd60724cb570302ba573790c779e8135722c4fc/docs/api/graphql/overview.md)
- [Getting Started](https://github.com/linkedin/datahub/blob/bbd60724cb570302ba573790c779e8135722c4fc/docs/api/graphql/getting-started.md)
- [Querying Metadata Entities](https://github.com/linkedin/datahub/blob/bbd60724cb570302ba573790c779e8135722c4fc/docs/api/graphql/querying-entities.md)

Also introduced auto-generated GraphQL Schema Reference documentation. These files are not checked in, but instead generated at build time. 

**Changes**

- Split monolithic GraphQL model file into 3 sub-schemas
        - entity.graphql: all CRUD operations against a given entity type (create, update, get, list)
        - search.graphql: all search and browse related graphql models
        - app.graphql: all UI specific graphql models (e.g. appConfig) 
- Added model descriptions for all GraphQL types and fields
- Deprecate misc Entity "info" fields for standardized "properties" and "editableProperties" fields
- Consistently accept urn, input arguments for each Update mutation to facilitate reuse of the Update input model in "create" APIs that will be coming soon. 
- Migrate "relationships" field input arguments to single wrapper "input" object, aligning with conventions across the app
- Add Dataset "properties" field, deprecate root-level property fields (ie. description) 
- Miscellaneous renaming of models for consistency
       - TagUpdateInput -> TagAssociationInput
       - TermUpdateInput -> TermAssociationInput
       - TagUpdate -> TagUpdateInput
       - PolicyUpdate -> PolicyUpdateInput
Breaking GraphQL API changes: 
- Remove the legacy string "tags" field from Dataset model
- Migrate custom properties for Dataset from root level "properties" field to new properties.customProperties to align with other entity convention 
- Added GraphQL API Documentation generation + added to documentation website 

Validated all changes locally.  

**Coming soon**

- Rich GraphQL Documentation on docs website
- addOwner, removeOwner mutations
- listEntity endpoints 
- createEntity, removeEntity endpoints 
- Consolidation in root-level key fields (urn, entity type, platform, name) 
Let us know which you'd like to see on our Slack :) 

## Checklist
- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable)

### CODE REVIEW

File: datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/types/dashboard/mappers/DashboardUpdateInputSnapshotMapper.java [hunk 1]
```diff
@@ -38,28 +36,32 @@ public DashboardSnapshot apply(@Nonnull final DashboardUpdateInput dashboardUpda
...
+                    )
+                );
+            } else {
+                // Tags override global tags
+                globalTags.setTags(
+                    new TagAssociationArray(
+                        dashboardUpdateInput.getTags().getTags().stream().map(
+                            element -> TagAssociationUpdateMapper.map(element)
+                        ).collect(Collectors.toList())
+                    )
```
Comment (gabe-lyons) [inline datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/types/dashboard/mappers/DashboardUpdateInputSnapshotMapper.java:62]: do we want legacy tags to appear if they are deprecated?

File: docs-website/graphql/generateGraphQLSchema.sh [hunk 2]
```diff
@@ -0,0 +1,8 @@
+#!/bin/sh
+rm combined.graphql
+touch combined.graphql
+echo "Generating combined GraphQL schema..."
+echo "# Auto Generated During Docs Build" >> combined.graphql
```
Comment (gabe-lyons) [inline docs-website/graphql/generateGraphQLSchema.sh:5]: why not just do a single `>` and drop the `rm`?

Comment (gabe-lyons) [reply docs-website/graphql/generateGraphQLSchema.sh:5]: just consistency w/ the lines below?

File: docs/api/graphql/querying-entities.md [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+As GraphQL:
+
+```graphql 
+{
+  search(input: { type: "DATASET", query: "my sql dataset", start: 0, count: 10 }) {
+    start
+    count
+    total
+    searchResults {
+      entity {
```
Comment (gabe-lyons) [inline docs/api/graphql/querying-entities.md:55]: maybe add the entity agnostic `urn` and `type` fields above `...on Dataset` so the user will see data even if they change entity type?

File: docs/api/graphql/querying-entities.md [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+
+For example, to add a Tag a DataFlow entity, you can issue the following GraphQL mutation: 
+
+*As GraphQL*
+
+```graphql 
+mutation addTag {
+    addTag(input: { tagUrn: "urn:li:tag:NewTag", resourceUrn: "urn:li:dataFlow:(airflow,dag_abc,PROD)" })
+}
+```
```
Comment (gabe-lyons) [inline docs/api/graphql/querying-entities.md:172]: 🙌

File: docs/api/graphql/querying-entities.md (hunk 3 above)
Comment (jjoyce0510) [reply docs/api/graphql/querying-entities.md:55]: good idea!

**Review by shirshanka — APPROVED** [review]
We finally have a public API! Really excited to see how the community responds.

**Pull request merged**